lib/kubeclient/common.rb

```diff
@@ -393,6 +401,30 @@ def patch_entity(resource_name, name, patch, namespace = nil)
       format_response(@as, response.body)
     end
 
+    def json_patch_entity(resource_name, name, patch, namespace = nil)
+      ns_prefix = build_namespace_prefix(namespace)
+      response = handle_exception do
+        rest_client[ns_prefix + resource_name + "/#{name}"]
+          .patch(
+            patch.to_json,
+            { 'Content-Type' => 'application/json-patch+json' }.merge(@headers)
+          )
+      end
+      format_response(@as, response.body)
+    end
+
+    def merge_patch_entity(resource_name, name, patch, namespace = nil)
+      ns_prefix = build_namespace_prefix(namespace)
+      response = handle_exception do
+        rest_client[ns_prefix + resource_name + "/#{name}"]
+          .patch(
+            patch.to_json,
+            { 'Content-Type' => 'application/merge-patch+json' }.merge(@headers)
+          )
+      end
+      format_response(@as, response.body)
+    end
```

Comment:
The considerations from #357 that led to separate public `*patch` methods don't apply for these internal undocumented methods.  Consider merging them into one method with 5 args.  (Also, you can make `namespace` mandatory).

Reply:
>you can make namespace mandatory

I'm surprised it wasn't to begin with since it would seem that [`build_ns_prefix(namespace)`](https://github.com/abonas/kubeclient/blob/master/lib/kubeclient/common.rb#L202) would crash in that case (called from[here](https://github.com/abonas/kubeclient/blob/master/lib/kubeclient/common.rb#L385)).

Making it mandatory is probably the best bet, since then it's really the responsibility of the caller at that point

Reply:
Sorry, I was unclear. Sure, for the public methods is must be optional.  
I meant on the `patch_entity` internal method, it's always passed a namespace arg, even if nil.  
(And by mandatory I meant at signature level, dropping the `= nil`, but still doing `build_namespace_prefix` supporting nil)

Never mind, fine either way.